### DISCUSSION TITLE
Easiest way to grab line and column of mouse click

### DISCUSSION THREAD

**Discussion by zuzu837** [sourcelair/xterm.js#4380] [Q&A]
Hello All, I have been searching for a way to grab x , y cursor position when user click in my xterm screen.
Seems that [https://github.com/xtermjs/xterm.js/issues/2336] was quite interesting but was not implemented.

As a last chance I did ask chatGPT and got an interesting answer:
```
const xterm = new Xterm();
xterm.attachCustomKeyEventHandler((event) => {
  const pos = xterm.getMouseBufferPosition(event);
  console.log('Ligne: ' + pos.y + ', Colonne: ' + pos.x);
});
```
Does not work ... but this is exactly what I am looking for ... maybe chatGPT is aware of 5.2 ?

**Comment by Tyriar** [comment]
It's a little tricky until https://github.com/xtermjs/xterm.js/issues/702 is implemented. The way VS Code does it is re-calculates the dimensions using similar logic (which is dangerous and will likely break) and then listens to regular mouse events on the xterm element.

**Comment by jerch** [comment]
You can work around missing official API by using private terminal methods:
```javascript
// some fake mouse event
const me = new MouseEvent('mousedown', {clientX: 100, clientY: 100});
// grab translated coords from private API Terminal._core._mouseService.getMouseReportCoords
// in demo:
const position = term._core._mouseService.getMouseReportCoords(me, term.element);
// --> {col: 10, row: 0, x: 90, y: 18}
```

Benefit here - it uses the same metrics the terminal would use internally for mouse handling, so no weird offset corrections to be done on your end.
